### PR TITLE
CoW Protocol: Adding new known-solver data

### PR DESCRIPTION
**Pull request by bh2smith**
Adding two new solver's metadata to the "known solvers" list. 

PoC Query is here: https://dune.com/queries/538257

Note that these solvers won't appear in the list until they have been formally registered and once this happens they will appear here with the added metadata.

### CODE REVIEW

**Comment by jeff-dude** [comment]
similar to last time, when dropping and rebuilding view_solvers, the view_batches need dropped and recreated too. i've paused both refresh cron jobs. i am working through dropping the views, but postgres is giving me troubles again. i'll get them recreated and turn the cron jobs on when it's working.

**Comment by bh2smith** [comment]
In the meantime. I will prepare a refactor of the view batches that will substantially reduce the runtime of the update job. Similar to what was done for the trades table in this recent PR. 

https://github.com/duneanalytics/abstractions/commit/3a8e5d3a881595b89924f9a79b821ffe92a4daf8

Once we have phased out all queries depending on this view we can completely remove it. In fact I think we have already eliminated the need for view trades.

**Comment by bh2smith** [comment]
Something else that come to mind is to remove the dependency of view batches on view solvers. I can always join these later to get the solver name column.

**Comment by bh2smith** [comment]
@jeff-dude - Here is a PR refactoring the batches so the job shouldn't continue consuming all your resources: #1015

once that is merged, I can work on migrating all our queries away from `view_batches` and delete it.

**Comment by jeff-dude** [comment]
> @jeff-dude - Here is a PR refactoring the batches so the job shouldn't continue consuming all your resources: #1015
> 
> once that is merged, I can work on migrating all our queries away from `view_batches` and delete it.

thank you for all the info and helping work through it. would you like me to wait on this PR then until #1015 is done? if so, i can start that one asap.

**Comment by bh2smith** [comment]
Yes please! We would also like to keep the view_batches running if possible. I could make another small PR removing the solver name from the `view_batches` so they are no longer codependent.

**Comment by jeff-dude** [comment]
> Yes please! We would also like to keep the view_batches running if possible. I could make another small PR removing the solver name from the `view_batches` so they are no longer codependent. But I won't be able to do that until tomorrow.

both views (solvers, batches) cron jobs have resumed while i work on #1015

**Comment by bh2smith** [comment]
You are a legend!

File: ethereum/gnosis_protocol_v2/view_solvers.sql [hunk 1]
```diff
@@ -32,6 +32,8 @@ known_solver_metadata (address, environment, name) as (
                  (decode('a6ddbd0de6b310819b49f680f65871bee85f517e', 'hex'), 'prod', 'Legacy'),
                  (decode('2d15894fac906386ff7f4bd07fceac43fcf80c73', 'hex'), 'prod', 'DexCowAgg'),
                  (decode('a97DEF4fBCBa3b646dd169Bc2eeE40f0f3fE7771', 'hex'), 'prod', 'BalancerSOR'),
+                 (decode('6fa201C3Aff9f1e4897Ed14c7326cF27548d9c35', 'hex'), 'prod', 'otex'),
+                 (decode('1e05EE95b8DfCB7C7567FA52e0ef47288b1C8Fb5', 'hex'), 'barn', 'otex'),
```
Comment (bh2smith) [inline ethereum/gnosis_protocol_v2/view_solvers.sql:36]:
```suggestion
                 (decode('109BF9E0287Cc95cc623FBE7380dD841d4bdEb03', 'hex'), 'barn', 'otex'),
```

Comment (fleupold) [reply ethereum/gnosis_protocol_v2/view_solvers.sql:36]: Looks good!

File: ethereum/gnosis_protocol_v2/view_solvers.sql [hunk 2]
```diff
@@ -32,6 +32,8 @@ known_solver_metadata (address, environment, name) as (
                  (decode('a6ddbd0de6b310819b49f680f65871bee85f517e', 'hex'), 'prod', 'Legacy'),
                  (decode('2d15894fac906386ff7f4bd07fceac43fcf80c73', 'hex'), 'prod', 'DexCowAgg'),
                  (decode('a97DEF4fBCBa3b646dd169Bc2eeE40f0f3fE7771', 'hex'), 'prod', 'BalancerSOR'),
+                 (decode('6fa201C3Aff9f1e4897Ed14c7326cF27548d9c35', 'hex'), 'prod', 'otex'),
+                 (decode('109BF9E0287Cc95cc623FBE7380dD841d4bdEb03', 'hex'), 'barn', 'otex'),
```
Comment (bh2smith) [inline ethereum/gnosis_protocol_v2/view_solvers.sql:36]:
```suggestion
                 (decode('6fa201C3Aff9f1e4897Ed14c7326cF27548d9c35', 'hex'), 'prod', 'Otex'),
                 (decode('109BF9E0287Cc95cc623FBE7380dD841d4bdEb03', 'hex'), 'barn', 'Otex'),
```

**Comment by bh2smith** [comment]
I was just taking a look at the cron schedule for this update and it looks like its every 12 hours (at noon and midnight). Perhaps there would be less issue if this was run just 30 seconds before.

**Review by jeff-dude — APPROVED** [review]
@bh2smith both solvers/batches views are up-to-date as of this PR and both cron jobs remain active. feel free to open new PR in future to remove usage of view_batches, similar to your other PR for view_trades

**Comment by bh2smith** [comment]
Thanks very much @jeff-dude - I am just wondering if we should already be seeing the two new solvers added here. when I query 

```
select * from gnosis_protocol_v2."view_solvers"
where environment = 'new'
```

they are still appearing here... Is there some chance to run the top of this file again...

**Comment by jeff-dude** [comment]
> Thanks very much @jeff-dude - I am just wondering if we should already be seeing the two new solvers added here. when I query
> 
> ```
> select * from gnosis_protocol_v2."view_solvers"
> where environment = 'new'
> ```
> 
> they are still appearing as Uncatalogued... Is there some chance to run the top of this file again?

i'm not sure why these are giving me trouble. let me give it another go. i can see the two new metadata values weren't added to materialized view definition. i'll take it step by step to ensure it's updating as expected:

1. pause view solvers & batches cron jobs
2. drop view batches
3. drop view solvers
4. validate both dropped and committed to postgres
5. rebuild view solvers
6. ensure as file suggests here
7. rebuild view batches
8. enable cron jobs

**Comment by bh2smith** [comment]
Those steps sound about right. Have you recently rebuilt the solver view? I am still seeing them as "new" perhaps I should make a function that runs the entire drop and rebuild on every run...

**Comment by bh2smith** [comment]
I am not actually sure what refresh materialized view actually does? I was expecting that it would run this content:

```
solver_activation_events as (
    select solver, evt_block_number, evt_index, True as activated
    from gnosis_protocol_v2."GPv2AllowListAuthentication_evt_SolverAdded"
    union
    select solver, evt_block_number, evt_index, False as activated
    from gnosis_protocol_v2."GPv2AllowListAuthentication_evt_SolverRemoved"
),
-- Sorting by (evt_block_number, evt_index) allows us to pick the most recent activation status of each unique solver
registered_solvers as (
    select distinct on (solver) solver,
                                activated as active
    from solver_activation_events
    order by solver, evt_block_number desc, evt_index desc
),
-- Manually inserting environment and name for each "known" solver
known_solver_metadata (address, environment, name) as (
    SELECT *
    FROM (VALUES (decode('f2d21ad3c88170d4ae52bbbeba80cb6078d276f4', 'hex'), 'prod', 'MIP'),
                 (decode('15f4c337122ec23859ec73bec00ab38445e45304', 'hex'), 'prod', 'ParaSwap'),
                 (decode('de1c59bc25d806ad9ddcbe246c4b5e5505645718', 'hex'), 'prod', '1inch'),
                 (decode('340185114f9d2617dc4c16088d0375d09fee9186', 'hex'), 'prod', 'Naive'),
                 (decode('833f076d182123ca8dde2743045ea02957bd61ef', 'hex'), 'prod', 'Baseline'),
                 (decode('e92f359e6f05564849afa933ce8f62b8007a1d5d', 'hex'), 'prod', '0x'),
                 (decode('77ec2a722c2393d3fd64617bbaf1499c713e616b', 'hex'), 'prod', 'QuasiModo'),
                 (decode('a6ddbd0de6b310819b49f680f65871bee85f517e', 'hex'), 'prod', 'Legacy'),
                 (decode('2d15894fac906386ff7f4bd07fceac43fcf80c73', 'hex'), 'prod', 'DexCowAgg'),
                 (decode('a97DEF4fBCBa3b646dd169Bc2eeE40f0f3fE7771', 'hex'), 'prod', 'BalancerSOR'),
                 (decode('6fa201C3Aff9f1e4897Ed14c7326cF27548d9c35', 'hex'), 'prod', 'Otex'),
                 (decode('109BF9E0287Cc95cc623FBE7380dD841d4bdEb03', 'hex'), 'barn', 'Otex'),
                 (decode('70f3c870b6e7e1d566e40c41e2e3d6e895fcee23', 'hex'), 'barn', 'QuasiModo'),
                 (decode('97dd6a023b06ba4722aF8Af775ec3C2361e66684', 'hex'), 'barn', '0x'),
                 (decode('6372bcbf66656e91b9213b61d861b5e815296207', 'hex'), 'barn', 'ParaSwap'),
                 (decode('158261d17d2983b2cbaecc1ae71a34617ae4cb16', 'hex'), 'barn', 'MIP'),
                 (decode('8c9d33828dace1eb9fc533ffde88c4a9db115061', 'hex'), 'barn', '1inch'),
                 (decode('bfaf2b5e351586551d8bf461ba5b2b5455b173da', 'hex'), 'barn', 'Baseline'),
                 (decode('b8650702412d0aa7f01f6bee70335a18d6a78e77', 'hex'), 'barn', 'Naive'),
                 (decode('583cD88b9D7926357FE6bddF0E8950557fcDA0Ca', 'hex'), 'barn', 'DexCowAgg'),
                 (decode('6c2999b6b1fad608ecea71b926d68ee6c62beef8', 'hex'), 'barn', 'Legacy'),
                 (decode('ED94b86275447e28ddbDd17BBEB1f62D607b5119', 'hex'), 'barn', 'Legacy'),
                 (decode('8ccc61dba297833dbe5d95fd6360f106b9a7576e', 'hex'), 'barn', 'Naive'),
                 (decode('0d2584da2f637805071f184bcfa1268ebae8a24a', 'hex'), 'barn', 'Baseline'),
                 (decode('a0044c620da7f2876da7004719b8370eb7be5e50', 'hex'), 'barn', 'MIP'),
                 (decode('da324c2f06d3544e7965767ce9ca536dcb67a660', 'hex'), 'barn', 'QuasiModo'),
                 (decode('e33062a24149f7801a48b2675ed5111d3278f0f5', 'hex'), 'barn', '1inch'),
                 (decode('080a8b1e2f3695e179453c5e617b72a381be44b9', 'hex'), 'barn', 'ParaSwap'),
                 (decode('de786877a10dbb7eba25a4da65aecf47654f08ab', 'hex'), 'barn', '0x'),
                 (decode('dae69affe582d36f330ee1145995a53fab670962', 'hex'), 'barn', 'DexCowAgg'),
                 (decode('0b78e29ee55aa73b366730cf512d65c514eeb196', 'hex'), 'barn', 'BalancerSOR'),
                 (decode('22dee0935c77d32c7241362b14e76fc2d5ef657d', 'hex'), 'barn', 'BalancerSOR'),
                 (decode('0798540ee03a8c2e68cef19c56d1faa86271d5cf', 'hex'), 'service', 'Withdraw'),
                 (decode('df30c9502eafea21ecc8410108dda338dd5047c5', 'hex'), 'service', 'Withdraw'),
                 (decode('256bb5ad3dbdf61ae08d7cbc0b9223ccb1c60aae', 'hex'), 'service', 'Withdraw'),
                 (decode('84e5c8518c248de590d5302fd7c32d2ae6b0123c', 'hex'), 'service', 'Withdraw'),
                 (decode('a03be496e67ec29bc62f01a428683d7f9c204930', 'hex'), 'service', 'Withdraw'),
                 (decode('2caef7f0ee82fb0abf1ab0dcd3a093803002e705', 'hex'), 'test', 'Test Solver 1'),
                 (decode('56d4ed5e49539ebb1366c7d6b8f2530f1e4fe753', 'hex'), 'test', 'Test Solver 2')
         ) as _
)
-- Combining the metadata with current activation status for final table
select solver as address,
       case when environment is not null then environment else 'new' end as environment,
       case when name is not null then name else 'Uncatalogued' end      as name,
       active
from registered_solvers
         left outer join known_solver_metadata
                         on solver = address;
```

but maybe I was mistaken...

**Comment by jeff-dude** [comment]
> > Thanks very much @jeff-dude - I am just wondering if we should already be seeing the two new solvers added here. when I query
> > ```
> > select * from gnosis_protocol_v2."view_solvers"
> > where environment = 'new'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > they are still appearing as Uncatalogued... Is there some chance to run the top of this file again?
> 
> i'm not sure why these are giving me trouble. let me give it another go. i can see the two new metadata values weren't added to materialized view definition. i'll take it step by step to ensure it's updating as expected:
> 
> 1. pause view solvers & batches cron jobs
> 2. drop view batches
> 3. drop view solvers
> 4. validate both dropped and committed to postgres
> 5. rebuild view solvers
> 6. ensure as file suggests here
> 7. rebuild view batches
> 8. enable cron jobs

through step 5 & view_solvers looks better. the query you shared doesn't return those two rows now

**Comment by bh2smith** [comment]
Thanks again! You are great!

**Comment by jeff-dude** [comment]
all the outlined steps are complete. let me know if the views are as expected. we can leave the cron jobs as refresh view for now if it works as expected.

**Pull request merged**